File: csharp/src/main/csharp/IO/Swagger/Model/ShippingInfo.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace IO.Swagger.Model
{
    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public partial class ShippingInfo :  IEquatable<ShippingInfo>
    { 
    
        /// <summary>
        /// Shipping status of the order item. It is set to 'Initial' when the order is submitted. Then it is changed to 'Accepted' when a distributor starts processing the order. Finally it is changed to Shipped which means that distributor has shipped the device.
        /// </summary>
        /// <value>Shipping status of the order item. It is set to 'Initial' when the order is submitted. Then it is changed to 'Accepted' when a distributor starts processing the order. Finally it is changed to Shipped which means that distributor has shipped the device.</value>
        [JsonConverter(typeof(StringEnumConverter))]
        public enum StatusEnum {
            
            [EnumMember(Value = "Initial")]
            Initial,
            
            [EnumMember(Value = "Accepted")]
            Accepted,
            
            [EnumMember(Value = "Shipped")]
            Shipped
        }

    
        /// <summary>
        /// Shipping status of the order item. It is set to 'Initial' when the order is submitted. Then it is changed to 'Accepted' when a distributor starts processing the order. Finally it is changed to Shipped which means that distributor has shipped the device.
        /// </summary>
        /// <value>Shipping status of the order item. It is set to 'Initial' when the order is submitted. Then it is changed to 'Accepted' when a distributor starts processing the order. Finally it is changed to Shipped which means that distributor has shipped the device.</value>
        [DataMember(Name="status", EmitDefaultValue=false)]
        public StatusEnum? Status { get; set; }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="ShippingInfo" /> class.
        /// Initializes a new instance of the <see cref="ShippingInfo" />class.
        /// </summary>
        /// <param name="Status">Shipping status of the order item. It is set to &#39;Initial&#39; when the order is submitted. Then it is changed to &#39;Accepted&#39; when a distributor starts processing the order. Finally it is changed to Shipped which means that distributor has shipped the device..</param>
        /// <param name="Carrier">Shipping carrier name. Appears only if the device status is \&quot;Shipped\&quot;.</param>
        /// <param name="TrackingNumber">Carrier-specific tracking number. Appears only if the device status is \&quot;Shipped\&quot;.</param>
        /// <param name="Method">Shipping method information.</param>
        /// <param name="Address">Shipping address for the order. If it coincides with the Emergency Service Address, then can be omitted. By default the same value as the emergencyServiceAddress. Multiple addresses can be specified; in case an order contains several devices, they can be delivered to different addresses.</param>

        public ShippingInfo(StatusEnum? Status = null, string Carrier = null, string TrackingNumber = null, List<ShippingMethod> Method = null, List<ShippingAddress> Address = null)
        {
            this.Status = Status;
            this.Carrier = Carrier;
            this.TrackingNumber = TrackingNumber;
            this.Method = Method;
            this.Address = Address;
            
        }
        
    
        /// <summary>
        /// Shipping carrier name. Appears only if the device status is \&quot;Shipped\&quot;
        /// </summary>
        /// <value>Shipping carrier name. Appears only if the device status is \&quot;Shipped\&quot;</value>
        [DataMember(Name="carrier", EmitDefaultValue=false)]
        public string Carrier { get; set; }
    
        /// <summary>
        /// Carrier-specific tracking number. Appears only if the device status is \&quot;Shipped\&quot;
        /// </summary>
        /// <value>Carrier-specific tracking number. Appears only if the device status is \&quot;Shipped\&quot;</value>
        [DataMember(Name="trackingNumber", EmitDefaultValue=false)]
        public string TrackingNumber { get; set; }
    
        /// <summary>
        /// Shipping method information
        /// </summary>
        /// <value>Shipping method information</value>
        [DataMember(Name="method", EmitDefaultValue=false)]
        public List<ShippingMethod> Method { get; set; }
    
        /// <summary>
        /// Shipping address for the order. If it coincides with the Emergency Service Address, then can be omitted. By default the same value as the emergencyServiceAddress. Multiple addresses can be specified; in case an order contains several devices, they can be delivered to different addresses
        /// </summary>
        /// <value>Shipping address for the order. If it coincides with the Emergency Service Address, then can be omitted. By default the same value as the emergencyServiceAddress. Multiple addresses can be specified; in case an order contains several devices, they can be delivered to different addresses</value>
        [DataMember(Name="address", EmitDefaultValue=false)]
        public List<ShippingAddress> Address { get; set; }
    
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class ShippingInfo {\n");
            sb.Append("  Status: ").Append(Status).Append("\n");
            sb.Append("  Carrier: ").Append(Carrier).Append("\n");
            sb.Append("  TrackingNumber: ").Append(TrackingNumber).Append("\n");
            sb.Append("  Method: ").Append(Method).Append("\n");
            sb.Append("  Address: ").Append(Address).Append("\n");
            
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as ShippingInfo);
        }

        /// <summary>
        /// Returns true if ShippingInfo instances are equal
        /// </summary>
        /// <param name="other">Instance of ShippingInfo to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(ShippingInfo other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.Status == other.Status ||
                    this.Status != null &&
                    this.Status.Equals(other.Status)
                ) && 
                (
                    this.Carrier == other.Carrier ||
                    this.Carrier != null &&
                    this.Carrier.Equals(other.Carrier)
                ) && 
                (
                    this.TrackingNumber == other.TrackingNumber ||
                    this.TrackingNumber != null &&
                    this.TrackingNumber.Equals(other.TrackingNumber)
                ) && 
                (
                    this.Method == other.Method ||
                    this.Method != null &&
                    this.Method.SequenceEqual(other.Method)
                ) && 
                (
                    this.Address == other.Address ||
                    this.Address != null &&
                    this.Address.SequenceEqual(other.Address)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                
                if (this.Status != null)
                    hash = hash * 59 + this.Status.GetHashCode();
                
                if (this.Carrier != null)
                    hash = hash * 59 + this.Carrier.GetHashCode();
                
                if (this.TrackingNumber != null)
                    hash = hash * 59 + this.TrackingNumber.GetHashCode();
                
                if (this.Method != null)
                    hash = hash * 59 + this.Method.GetHashCode();
                
                if (this.Address != null)
                    hash = hash * 59 + this.Address.GetHashCode();
                
                return hash;
            }
        }

    }
}

File: javascript/src/model/ShippingInfo.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient', './ShippingAddress', './ShippingMethod'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./ShippingAddress'), require('./ShippingMethod'));
  } else {
    // Browser globals (root is window)
    if (!root.RingCentralApi) {
      root.RingCentralApi = {};
    }
    root.RingCentralApi.ShippingInfo = factory(root.RingCentralApi.ApiClient, root.RingCentralApi.ShippingAddress, root.RingCentralApi.ShippingMethod);
  }
}(this, function(ApiClient, ShippingAddress, ShippingMethod) {
  'use strict';

  /**
   * The ShippingInfo model module.
   * @module model/ShippingInfo
   * @version v1.0
   */

  /**
   * Constructs a new <code>ShippingInfo</code>.
   * @alias module:model/ShippingInfo
   * @class
   */
  var exports = function() {






  };

  /**
   * Constructs a <code>ShippingInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ShippingInfo} obj Optional instance to populate.
   * @return {module:model/ShippingInfo} The populated <code>ShippingInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('status')) {
        obj['status'] = ApiClient.convertToType(data['status'], 'String');
      }
      if (data.hasOwnProperty('carrier')) {
        obj['carrier'] = ApiClient.convertToType(data['carrier'], 'String');
      }
      if (data.hasOwnProperty('trackingNumber')) {
        obj['trackingNumber'] = ApiClient.convertToType(data['trackingNumber'], 'String');
      }
      if (data.hasOwnProperty('method')) {
        obj['method'] = ApiClient.convertToType(data['method'], [ShippingMethod]);
      }
      if (data.hasOwnProperty('address')) {
        obj['address'] = ApiClient.convertToType(data['address'], [ShippingAddress]);
      }
    }
    return obj;
  }


  /**
   * Shipping status of the order item. It is set to 'Initial' when the order is submitted. Then it is changed to 'Accepted' when a distributor starts processing the order. Finally it is changed to Shipped which means that distributor has shipped the device.
   * @member {module:model/ShippingInfo.StatusEnum} status
   */
  exports.prototype['status'] = undefined;

  /**
   * Shipping carrier name. Appears only if the device status is \"Shipped\"
   * @member {String} carrier
   */
  exports.prototype['carrier'] = undefined;

  /**
   * Carrier-specific tracking number. Appears only if the device status is \"Shipped\"
   * @member {String} trackingNumber
   */
  exports.prototype['trackingNumber'] = undefined;

  /**
   * Shipping method information
   * @member {Array.<module:model/ShippingMethod>} method
   */
  exports.prototype['method'] = undefined;

  /**
   * Shipping address for the order. If it coincides with the Emergency Service Address, then can be omitted. By default the same value as the emergencyServiceAddress. Multiple addresses can be specified; in case an order contains several devices, they can be delivered to different addresses
   * @member {Array.<module:model/ShippingAddress>} address
   */
  exports.prototype['address'] = undefined;


  /**
   * Allowed values for the <code>status</code> property.
   * @enum {String}
   * @readonly
   */
  exports.StatusEnum = { 
    /**
     * value: Initial
     * @const
     */
    INITIAL: "Initial",
    
    /**
     * value: Accepted
     * @const
     */
    ACCEPTED: "Accepted",
    
    /**
     * value: Shipped
     * @const
     */
    SHIPPED: "Shipped"
  };

  return exports;
}));
